pkg/catalog/routes.go

```diff
@@ -514,9 +529,150 @@ func (mc *MeshCatalog) listMeshServices() []service.MeshService {
 	return services
 }
 
+func (mc *MeshCatalog) destinationServicesFromTrafficTarget(t *target.TrafficTarget) ([]service.MeshService, error) {
+	sa := service.K8sServiceAccount{
+		Name:      t.Spec.Destination.Name,
+		Namespace: t.Spec.Destination.Namespace,
+	}
+	destServices, err := mc.GetServicesForServiceAccount(sa)
+	if err != nil {
+		return nil, errors.Errorf("Err finding Services for Service Account %#v: %v", sa, err)
+	}
+	return destServices, nil
+}
+
+func (mc *MeshCatalog) getHostnamesBasedOnNamespace(meshService service.MeshService) ([]string, []string, error) {
```

Comment:
Could you add a comment to this function given the function name has `BasedOnNamespace` but there is no parameter for a namespace.